server/src/main/java/com/google/musicanalysis/api/perspective/PerspectiveRequest.java

```diff
@@ -0,0 +1,77 @@
+package com.google.musicanalysis.site;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+
+public class PerspectiveRequest {
+
+  private String text;
+  private ArrayList<String> attributes;
+
+  public PerspectiveRequest(String text, ArrayList<String> attributes) {
+    this.text = text;
+    this.attributes = attributes;
+  }
+
+  /**
+   * Sends out request JSON Object and returns the JSON response from Perspective API.
+   *
+   * @return the JSON response as string
+   */
+  public String getResponse() throws MalformedURLException, IOException {
+
+    String jsonString = buildJson(this.text, this.attributes);
+    URL url = new URL("https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=");
+
+    // Open up the connection
+    HttpURLConnection con = (HttpURLConnection) url.openConnection();
+    con.setRequestMethod("POST");
+    con.setRequestProperty("Content-Type", "application/json; utf-8");
+    con.setRequestProperty("Accept", "application/json");
+    con.setDoOutput(true);
```

Comment:
Oh nice I used the `com.google.musicanalysis.util.URLEncodedBuilder` , `java.net.http.HttpClient` , and `java.net.http.HttpRequest` like ibiyemi but `java.net.URL` and `java.net.HttpURLConnection` seems to be doing the job and is much more readable 👍